system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -1968,16 +1944,22 @@ class CLdapClient : implements ILdapClient, public CInterface
                     LDAP_UNBIND(user_ld);
                 }
                 DBGLOG("finished LdapBind for user %s, rc=%d", username, rc);
-                if(!LdapServerDown(rc) || retries > LDAPSEC_MAX_RETRIES)
+
+                if(rc==LDAP_SERVER_DOWN || rc==LDAP_UNAVAILABLE)
+                {
+                    m_ldapconfig->rejectHost(hostbuf);
+                    continue;//try again with next configured LDAP host
+                }
+                else if(rc==LDAP_TIMEOUT && retries < LDAPSEC_MAX_RETRIES)
```

Comment:
Same comment as above. Curious why the test on retries here and not let the loop exit test handle it.

Reply:
Same response as above.